### DISCUSSION TITLE
[CRUD Service] array fields not filtered in array of objects with JSON Schema

### DISCUSSION THREAD

**Discussion by flower-of-the-bridges** [mia-platform/community#158] [Q&A] [answered]
Hi!

I've defined a crud data model in the `single view`/`mongodb crud` section, where there is a field called `locations` that is an array of objects.

In the details section, I've defined the following JSON Schema:
```
{
  "type": "object",
  "properties": {
    "name": { "type": "string" },
    "population": { "type": "number" }
  },
  "additionalProperties": false
}
```
However, If a record on Mongo has the following  item inside the location array: 
```
{
  // ...
  "locations": [{
     "name": "Milano",
     "population": 1325000,
     "region": "Lombardia"
   }]
}
```
the crud service returns after a GET request the array location with the field `region` in it. However, since I've set the field `additionalProperties`: `false`, the property either should not be shown or the validation should throw an error.

What am I missing?

**Comment by ThisIsDemetrio** [comment]
Hi @giovf96.

It could be a bug since, of course, `additionalProperties` set to `false` means you shouldn't see the fields not included in the Schema.

If you're working with the Console, can you give a look at the config map generated (from the Repository, inside the folder _config-maps/crud-service_) to verify if the JSON schema generated is the same expected from your configuration?

Also, may I ask you what version of the CRUD Service are you using?

**Reply by flower-of-the-bridges** [reply]
Hi @ThisIsDemetrio 

Thanks for the suggestion!

Currently I'm using `crud-service:v6.1.3`. If I see the `config-map` of the collection, the field `locations` has the following structure:

```
{
      "name": "locations",
      "description": "Array di locations",
      "type": "Array",
      "required": false,
      "nullable": false,
      "items": {
        "type": "RawObject",
        "schema": {
          "properties": {
             "name": { "type": "string" },
             "population": { "type": "number" }
           },
           "additionalProperties": false
         }
      }
}
``` 
From what I see, the JSON Schema has been placed correctly in the `schema` field of `locations`, however the `type: "object"` has been omitted and changed to the `RawObject` one.

**Comment by ThisIsDemetrio** [comment, marked as answer]
It sounds like a bug to me, since this should work fine. I've tried myself and it looks it works fine with the GET /:id requests, not the GET / requests (which return a list).

We're opening an issue about it and try to find a quick solution to it.

**Reply by flower-of-the-bridges** [reply]
Thanks @ThisIsDemetrio
